### DISCUSSION TITLE
impala 语法解析器，使用 antlr4 适配 golang 语言，发生了异常

### DISCUSSION THREAD

**Discussion by HyDreamFly** [DTStack/dt-sql-parser#321] [Q&A]
antlr4 版本：4.13.1
生成代码结构：
![image](https://github.com/DTStack/dt-sql-parser/assets/73807082/2013af00-e827-4178-96fc-a37e557c4b4b)
异常位置：
![image](https://github.com/DTStack/dt-sql-parser/assets/73807082/d7ffb020-8af4-422b-932a-4ffd9f5a97af)
感觉发生问题的原因应该是在 ImpalaSqlParser.g4 这个文件：
![image](https://github.com/DTStack/dt-sql-parser/assets/73807082/c7f5c925-868c-4f58-9ee5-d2c287ab83fb)
有没有解决方案呢？对 antlr4 语法解析器不是很懂

**Comment by HaydenOrz** [comment]
Hi, 报歉这么晚才看到。

@header 是 Antlr4 的一种 acrtion ，在 @header {}  内部的代码会被原封不动的生成在 “生成文件” 的顶部。由于 dt-sql-parser 是 typescript 版本，这里的 import 是 typescript 模块倒导入语法。如果你不需要 ImpalaSqlParser 继承自某个基类，你可以移除

 1. @header
    ```js
    @header {
    import { SQLParserBase } from '../SQLParserBase';
    }
    ```
2. options 中的
      ```
      superClass=SQLParserBase
      ```
4. columnNamePath 规则中的
      ```
        | {this.shouldMatchEmpty()}?
      ```

然后重新尝试从 g4 文件中生成。
